python/GafferOSLUI/OSLObjectUI.py

```diff
@@ -41,6 +42,172 @@
 
 import GafferOSL
 
+import imath
+import functools
+
+# TODO _ duplicated from CompoundDataPlugValueWidget._MemberPlugValueWidget, omitted support for enabled
+class _MemberPlugValueWidget( GafferUI.PlugValueWidget ) :
+
+	def __init__( self, childPlug ) :
+
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 )
+
+		GafferUI.PlugValueWidget.__init__( self, self.__row, childPlug )
+
+		if not childPlug.getFlags( Gaffer.Plug.Flags.Dynamic ) :
```

Comment:
I just wasn't sure how much to clean this up.  If we actually want this to be fully identical to CompoundDataPlugValueWidget._MemberPlugValueWidget, should we just make that public?  Or do I just duplicate the  same code across CompoundDataPlugValueWidget, OSLObjectUI, and OSLImageUI, with the only difference being checking this dynamic flag?

Reply:
For now I would say just duplicate the code rather than dive into trying to generalise things. I think what we might want to do is just give the LayoutPlugValueWidget a horizontal layout mode, and allow metadata to be used to fix the width of certain widgets.